### PR TITLE
Sal 18 subscriptions endpoint

### PR DESCRIPTION
**Pull request by ATripathy1**
@lsaffie something related to the routing is not working. Have to find out the proper documentation that relates to the routing issues on the tests.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class ApplicationController < ActionController::API
 
   include ActionController::HttpAuthentication::Token::ControllerMethods
 
-  before_action :authenticate!
+  # before_action :authenticate!
```
Comment (lsaffie) [inline app/controllers/application_controller.rb]: gotta mock this call

File: app/controllers/v1/company_subscriptions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class V1::CompanySubscriptionsController < ApplicationController
+  before_action :set_subscription, only: [:update]
+  before_action :set_company, only: [:create]
```
Comment (lsaffie) [inline app/controllers/v1/company_subscriptions_controller.rb]: get_company is a more suited name

File: app/controllers/v1/company_subscriptions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  def set_company
+    @company = Company.find(subscription_params[:company_id])
+  end
+
+  def set_subscription
+    @subscription = Subscription.find(params[:id])
+  end
+
+  def subscription_params
+    params.require(:subscription).permit(:external_id, :company_id, :begin_at, :end_at, :no_of_categories, :no_of_service_areas, :state)
```
Comment (lsaffie) [inline app/controllers/v1/company_subscriptions_controller.rb]: you'll have to drop the external_id and company_id from here

File: app/controllers/v1/company_subscriptions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+class V1::CompanySubscriptionsController < ApplicationController
```
Comment (lsaffie) [inline app/controllers/v1/company_subscriptions_controller.rb]: huh? Why CompanySubscriptionsController?

File: db/migrate/20150615181817_add_external_id_to_subscriptions.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddExternalIdToSubscriptions < ActiveRecord::Migration
+  def change
```
Comment (lsaffie) [inline db/migrate/20150615181817_add_external_id_to_subscriptions.rb]: don't need this

**Comment by ATripathy1** [comment]
@lsaffie please review.

**Pull request merged**